src/manifest_store.rs

```diff
@@ -29,9 +30,26 @@ pub(crate) async fn apply_db_state_update<F>(
 ) -> Result<(), SlateDBError>
 where
     F: Fn(&StoredManifest) -> Result<CoreDbState, SlateDBError>,
+{
+    maybe_apply_db_state_update(manifest, |stored_manifest| {
+        let mutated_state = mutator(stored_manifest)?;
+        Ok(Some(mutated_state))
+    })
+    .await
+}
+
+pub(crate) async fn maybe_apply_db_state_update<F>(
```

Comment:
why is this called `maybe_apply`? Does it loop until it is able to update, or if it hits an error?

Reply:
If the mutator returns None, then no update is applied.